src/test_archive.cpp

```diff
@@ -0,0 +1,154 @@
+#include "engine/engine.h"
+#include "engine/archive.h"
+#include <boost/program_options.hpp>
+
+#include <string>
+#include <iostream>
+#include <cstdlib>
+#include <functional>
+#include <chrono>
+
+using namespace CityFlow;
+namespace bpo = boost::program_options;
+constexpr int interval = 100;
+
+double getDuration(const clock_t &a, const clock_t &b) {
+    return ((double)(a - b)) / CLOCKS_PER_SEC;
+}
```

Comment:
This timer method calculates CPU time, which means if we use multi-threading, the result will be several times larger than real time.